crates/hyperswitch_connectors/src/connectors/jpmorgan/transformers.rs

```diff
@@ -56,84 +140,245 @@ impl TryFrom<&JpmorganRouterData<&PaymentsAuthorizeRouterData>> for JpmorganPaym
         }
     }
 }
-// PaymentsResponse
-//TODO: Append the remaining status flags
-#[derive(Debug, Clone, Default, Serialize, Deserialize, PartialEq)]
-#[serde(rename_all = "lowercase")]
-pub enum JpmorganPaymentStatus {
-    Succeeded,
-    Failed,
+
+#[derive(Clone, Debug, Serialize, Deserialize, PartialEq)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum JpmorganTransactionStatus {
+    Success,
+    Denied,
+    Error,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize, Clone)]
+#[serde(rename_all = "UPPERCASE")]
+pub enum JpmorganTransactionState {
+    Closed,
+    Authorized,
+    Voided,
     #[default]
-    Processing,
+    Pending,
+    Declined,
+    Error,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct JpmorganPaymentsResponse {
+    transaction_id: String,
+    request_id: Option<String>,
+    transaction_state: JpmorganTransactionState,
+    response_status: String,
+    response_code: String,
+    response_message: Option<String>,
+    payment_method_type: Option<PaymentMethodType>,
+    capture_method: Option<CapMethod>,
+    is_capture: Option<bool>,
+    initiator_type: Option<String>,
+    account_on_file: Option<String>,
+    transaction_date: Option<String>,
+    approval_code: Option<String>,
+    host_message: Option<String>,
+    amount: Option<i64>,
+    currency: Option<common_enums::Currency>,
+    remaining_refundable_amount: Option<i64>,
+    remaining_auth_amount: Option<i64>,
+    host_reference_id: Option<String>,
+    merchant: Option<Merchant>,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Merchant {
+    merchant_id: Option<String>,
+    merchant_software: MerchantSoftware,
+    merchant_category_code: Option<String>,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct MerchantSoftware {
+    company_name: String,
+    product_name: String,
+    version: Option<String>,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct PaymentMethodType {
+    card: Option<Card>,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct Card {
+    expiry: Option<ExpiryResponse>,
+    card_type: Option<String>,
+    card_type_name: Option<String>,
+    masked_account_number: Option<String>,
+    card_type_indicators: Option<CardTypeIndicators>,
+    network_response: Option<NetworkResponse>,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct NetworkResponse {
+    address_verification_result: Option<String>,
+    address_verification_result_code: Option<String>,
+    card_verification_result_code: Option<String>,
+}
+
+#[derive(Default, Debug, Serialize, Deserialize)]
+#[serde(rename_all = "camelCase")]
+pub struct ExpiryResponse {
+    month: Option<i32>,
+    year: Option<i32>,
```

Comment:
Can we wrap this within a `Secret`? This would otherwise log these values in plaintext...